src/components/BaseNavbar.vue

```diff
@@ -104,6 +107,11 @@ export default {
     };
   },
   methods: {
+    debounce,
+    debouncedSearch: debounce(
```

Comment:
let's reduce the time to 500ms and myabe this can be transformed into an arrow function